### PR TITLE
scio: remove `ScioContext` from `result()`

### PR DESCRIPTION
**Pull request by danielnorberg**
If the scio job is asynchronously executed by some alternative flo runtime, the `ScioContext` instance used to submit the job might not exist any more when the job completes.

Note that this is a breaking api change.

### CODE REVIEW

**Comment by honnix** [comment]
So will it be a `null` or?

**Comment by danielnorberg** [comment]
@honnix no, not `null`. No `ScioContext` argument at all.

**Comment by honnix** [comment]
Ok I understand.

**Comment by honnix** [comment]
I meant before this PR. SC would refer to something not exist, is that right?

**Comment by honnix** [comment]
I'm ok with this breaking change, but maybe let's send PRs to intetnal users. Wouldn't be many.

**Comment by danielnorberg** [comment]
Yeah a runtime that wants to call the existing `result` method would have to pass in either `null` or a fake `ScioContext` if it doesn't have the original one anymore.

**Comment by danielnorberg** [comment]
Yeah, we can send PRs to users.

https://codesearch.spotify.net/search?q=%5C.result%5C%28.%2B%3D%3E&path=%5C.scala&rl=250

File: contrib/flo-scio_2.12/src/main/scala/com/spotify/flo/contrib/scio/ScioOperator.scala [hunk 1]
```diff
@@ -80,7 +80,7 @@ class ScioOperator[T] extends TaskOperator[ScioJobSpec.Provider[T], ScioJobSpec[
       }
 
       // Handle result
-      val result = Try(spec.result(sc, scioResult.get))
+      val result = Try(spec.result(scioResult.get))
```
[the remark this answers is not in the document]
Comment (danielnorberg) [reply contrib/flo-scio_2.12/src/main/scala/com/spotify/flo/contrib/scio/ScioOperator.scala]: The user can create the options if needed though.

[the remark this answers is not in the document]
Comment (honnix) [reply contrib/flo-scio_2.12/src/main/scala/com/spotify/flo/contrib/scio/ScioOperator.scala]: That seems weird and creates a good opportunity for inconsistency...

[the remark this answers is not in the document]
Comment (honnix) [reply contrib/flo-scio_2.12/src/main/scala/com/spotify/flo/contrib/scio/ScioOperator.scala]: PTAL #164

[the remark this answers is not in the document]
Comment (danielnorberg) [reply contrib/flo-scio_2.12/src/main/scala/com/spotify/flo/contrib/scio/ScioOperator.scala]: It seems weird for the flo execution environment to have to persist the options.

**Comment by honnix** [comment]
@danielnorberg What do you think if we change `ScioResult, PipelineOptions` to `PipelineOptions, ScioResult` so that in most cases no user change is required if they are not using `ScioContext`? Or would we prefer breaking loudly?

**Comment by honnix** [comment]
According to codesearch, there isn't any real usage of ScioJobSpec, and for those existing usages people just ignore ScioContext, so I would prefer we swap the order.

**Comment by honnix** [comment]
codecov is back, magically.

**Comment by honnix** [comment]
Strange codecov complains the delta change. Those lines are covered.

**Pull request merged**